src/runtime/tests/manifest-parser-test.ts

```diff
@@ -538,6 +538,76 @@ describe('manifest parser', () => {
         )
     `);
   });
+  it('parses type variables without constraints', () => {
+    parse(`
+      particle Foo
+        data: reads ~a
+    `);
+  });
+  it('parses type variables with a constraint', () => {
+    parse(`
+      particle Foo
+        data: reads ~a with {name: Text}
+    `);
+  });
+  it('parses type variables with multiple constraints', () => {
+    parse(`
+      particle Foo
+        data: reads ~a with {name: Text, age: Number}
+    `);
+  });
+  it('parses max type variables without constraints', () => {
+    parse(`
+      particle Foo
+        data: reads ~a with {*}
```

Comment:
Can we have a test for complaining if you do:
```
particle Foo
  data: reads {*}
```

Reply:
This is implemented in `manifest-test.ts`